lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -627,6 +596,83 @@ function writeStringsToCtcFiles(locale, strings) {
   fs.writeFileSync(fullPath, JSON.stringify(output, null, 2) + '\n');
 }
 
+/**
+ * @param {Record<string, CtcMessage>} strings
+ */
+function checkForCollisions(strings) {
+  /** @type {Map<string, string>} */
+  const seenStrings = new Map();
+
+  /** @type {Array<string>} */
+  const collisionStrings = [];
+
+  // check for duplicates, if duplicate, add @description as @meaning to both
+  for (const [messageKey, ctc] of Object.entries(strings)) {
+    if (seenStrings.has(ctc.message)) {
+      ctc.meaning = ctc.description;
+      const seenId = seenStrings.get(ctc.message);
+      // TODO: `strings[seenId]` check shouldn't be necessary here ...
+      // see https://github.com/GoogleChrome/lighthouse/pull/12441/files#r630521367
+      if (seenId && strings[seenId]) {
+        if (!strings[seenId].meaning) {
+          strings[seenId].meaning = strings[seenId].description;
+          collisionStrings.push(ctc.message);
```

Comment:
I added this at @patrickhulce's suggestion. tbh I don't understand what's going on here. the old code was doing `collisions++` in both places. should I just do one `collisionStrings.push(ctc.message);` here (keep L626?)

Reply:
> I added this at @patrickhulce's suggestion. tbh I don't understand what's going on here. the old code was doing `collisions++` in both places. should I just do one `collisionStrings.push(ctc.message);` here (keep L626?)

it's the difference between the number of strings with collisions (which is the current count) vs the number of pairs of strings that collide with each other